projects/auth0-angular/src/lib/auth.service.ts

```diff
@@ -28,58 +29,87 @@ import {
   filter,
   takeUntil,
   distinctUntilChanged,
+  catchError,
+  switchMap,
 } from 'rxjs/operators';
 
 import { Auth0ClientService } from './auth.client';
 import { WindowService } from './window';
 import { AbstractNavigator } from './abstract-navigator';
+import { connectableObservableDescriptor } from 'rxjs/internal/observable/ConnectableObservable';
 
 @Injectable({
   providedIn: 'root',
 })
 export class AuthService implements OnDestroy {
   private isLoadingSubject$ = new BehaviorSubject(true);
   private isAuthenticatedSubject$ = new BehaviorSubject(false);
+  private errorSubject$ = new ReplaySubject<Error>(1);
```

Comment:
what's the `1` for? any reason to not use BehaviorSubject here? on the template, you refer to this as "last error". Sounds like we should be keeping **only** the last value only -> behavior subject

Reply:
The `1` is the buffer size, which means that it only gives you the last value.

The difference is that `ReplaySubject` does not have an initial value, whereas `BehaviorSubject` does. In this case, I feel like the default state for the error observable is to have nothing and emit nothing, until an error occurs.

https://rxjs-dev.firebaseapp.com/api/index/class/ReplaySubject